### DISCUSSION TITLE
Type for fields with default value

### DISCUSSION THREAD

**Discussion by dawidmachon** [payloadcms/payload#2580] [Q&A]
Hi,
I have created collection `Product` with field:
```ts
{
      name: "price",
      type: "number",
      defaultValue: 0
    },
```
type which is generated for it by payload:
`Product.price?: number | undefined`

I would ask if there is any config option to check and enforce strong typing. When you have a default value, then it will always be defined, but now payload thinks differently.

**Comment by DanRibbens** [comment]
That is a good question. 

If you're reading that `Product` from a `find` query, of course the defaultValue is set so the property doesn't need to be marked `?` optional. The trick with dropping optional is that when the type is used to `create`, it isn't required.

Since we don't have permutations for types such as CreateProduct vs ReadProduct, we just have to always mark them optional. I don't think this is a good direction to go. 

I'm interested in knowing if you have another solution to this problem.

**Reply by DanRibbens** [reply]
~I have changed my mind on this already and now agree that defaultValue should make the type required.~ Nevermind, this would break the `data` argument of payload.create().

~I see that the createdAt and updatedAt timestamp fields are set to required in the type generation. My example of `create` doesn't matter because the `data` is a Partial of the collection type and so required or not doesn't matter anyways.~
We do make timestamps optional for `data` in the create options and I'm not how we could do that for all defaultValue fields.
